lib/apress/utils/extensions/active_record/validate_uniqueness_in_memory.rb

```diff
@@ -11,20 +11,38 @@ def self.included(model)
   module ValidateMethods
     def self.included(model)
       model.class_eval do
-        # Validate that the the objects in +collection+ are unique
-        # when compared against all their non-blank +attrs+. If not
-        # add +message+ to the base errors.
-        def validate_uniqueness_of_in_memory(collection, attrs, message, attribute = :base)
+        # Validate that the objects in collection are unique when compared against all their non-blank attrs.
+        #   If not add message to the base errors.
+        #
+        # options:
+        #   assign_error_to_collection - assign error message to each model in collection
+        #   additional_message - set additional message to :base error for a current model
+        #
+        # Returns Object
+        def validate_uniqueness_of_in_memory(collection, attrs, message, attribute = :base, options = {})
```

Comment:
Metrics/CyclomaticComplexity: Cyclomatic complexity for validate_uniqueness_of_in_memory is too high. [10/6]<br>Metrics/PerceivedComplexity: Perceived complexity for validate_uniqueness_of_in_memory is too high. [10/7]